maven.gradle

```diff
@@ -1,27 +1,44 @@
 def properties = new Properties()
 def propsFile = file("$rootDir/publish.properties")
+
+/*
+* the file will not exist if you go with system based sonatype setup - Option 3
+* */
 if (propsFile.exists()) {
     properties.load(propsFile.newInputStream())
 }
 
 /*
-* 1. findProperty("sonatypeUsername"): use parameter from the command (if attached) so you can run ->
-* /gradlew deploy -PsonatypeUsername={yourMavenUsername} -PsonatypePassword={yourMavenPassword}.
-* 2. properties["sonatype.user"]: use property defined in publish.properties file in the root directory. If you have that file in the root
-* of the project you can run -> ./gradlew deploy
-* 3. System.getenv("SONATYPE_PASS"): use global variable defined in the system. No need to create publish.properties file, you can simply
-* run -> ./gradlew deploy.
-*/
+* There are 3 possible approaches to retrieving previously defined sonatype credentials and setting it on a sonatype file:
+*
+*    Option 1:
+*       - findProperty("SOME_PROPERTY") - use parameter from the command (if attached) so you can run ->
+*         ./gradlew deploy -PsonatypeSOME_PROPERTY_NAME={SOME_PROPERTY_VALUE} -PsonatypeSOME_PROPERTY_NAME_2={SOME_PROPERTY_VALUE_2}
+*
+*    Option 2:
+*       - properties["SOME_PROPERTY"]: use property defined in publish.properties file in the root directory(if going with this option the
+*         file must first be manually created as described in the handbook) so you can run -> ./gradlew deploy
+*
+*    Option 3:
+*       - System.getenv("SONATYPE_PASS"): use global variable defined in the system. With this approach there is no need to create a
+*         publish.properties file so you can simply run -> ./gradlew deploy
+* */
```

Comment:
I do wonder if we should even present so many options or if we should just decide on one as the recommended one and go for it 🤔 I'm pretty sure that option 2 is the simplest one.

Reply:
Most probably the simplest one but wanted to provide the users with options since I did use Option 3 for JsonApiX. This was actually the motivation behind the current improvement but we can probably discuss it more.